OpenRA.Mods.Common/Traits/Render/WithChargeOverlay.cs

```diff
@@ -44,14 +47,27 @@ public WithChargeOverlay(ActorInitializer init, WithChargeOverlayInfo info)
 
 			overlay = new Animation(init.World, renderSprites.GetImage(init.Self));
 
-			renderSprites.Add(new AnimationWithOffset(overlay, null, () => !charging),
-				info.Palette, info.IsPlayerPalette);
+			// HACK: WithChargeOverlay added before sprite body.
+			init.World.AddFrameEndTask(w =>
+				renderSprites.Add(new AnimationWithOffset(overlay, null, () => !charging),
```

Comment:
Use [this constructor](https://github.com/OpenRA/OpenRA/blob/12966129a4e412fd2c53eb05b310f30557058521/OpenRA.Game/Graphics/AnimationWithOffset.cs#L30) instead and set the `zOffset`. That should take care of the issue you are working around with this.